dizoo/gfootball/envs/gfootball_academy_env.py

```diff
@@ -0,0 +1,318 @@
+                ego_position = full_obs['left_team'][-self.n_agents + index].reshape(-1)
+                simple_obs.append(ego_position)
+                simple_obs.append(
+                    (np.delete(full_obs['left_team'][-self.n_agents:], index, axis=0) - ego_position).reshape(-1)
+                )
+
+                simple_obs.append(full_obs['left_team_direction'][-self.n_agents + index].reshape(-1))
+                simple_obs.append(
+                    np.delete(full_obs['left_team_direction'][-self.n_agents:], index, axis=0).reshape(-1)
+                )
+
+                simple_obs.append(full_obs['right_team'][0] - ego_position)
+                simple_obs.append(full_obs['right_team'][1] - ego_position)
+                simple_obs.append(full_obs['right_team'][2] - ego_position)
+                simple_obs.append(full_obs['right_team_direction'][0])
+                simple_obs.append(full_obs['right_team_direction'][1])
+                simple_obs.append(full_obs['right_team_direction'][2])
+
+                simple_obs.append(full_obs['ball'][:2] - ego_position)
+                simple_obs.append(full_obs['ball'][-1].reshape(-1))
+                simple_obs.append(full_obs['ball_direction'])
+
+        simple_obs = np.concatenate(simple_obs)
+        return simple_obs
+
+    def get_global_state(self):
+        return self.get_simple_obs(-1)
+
+    def get_global_special_state(self):
+        return [np.concatenate([self.get_global_state(), self.get_obs_agent(i)]) for i in range(self.n_agents)]
+
+    def check_if_done(self):
+        cur_obs = self._env.unwrapped.observation()[0]
+        ball_loc = cur_obs['ball']
+        ours_loc = cur_obs['left_team'][-self.n_agents:]
+
+        if ball_loc[0] < 0 or any(ours_loc[:, 0] < 0):
+            return True  # TODO(pu)
+
+        return False
+
+    def reset(self):
+        """Returns initial observations and states."""
+        self.time_step = 0
+        self._env.reset()
+        obs = {
+            'agent_state': np.stack(self.get_obs(), axis=0).astype(np.float32),
+            # Note: here 'global_state' is the agent_specific_global_state,
+            # we simply concatenate the global_state and agent_state
+            'global_state': np.stack(
+                self.get_global_special_state(),
+                axis=0,
+            ).astype(np.float32),
+            'action_mask': np.stack(self.get_avail_actions(), axis=0).astype(np.float32),
+        }
+
+        if hasattr(self, '_seed') and hasattr(self, '_dynamic_seed') and self._dynamic_seed:
+            np_seed = 100 * np.random.randint(1, 1000)
+            self._env.seed(self._seed + np_seed)
+        elif hasattr(self, '_seed'):
+            self._env.seed(self._seed)
+        self._final_eval_reward = 0
+
+        return obs
+
+    def step(self, actions):
+        """Returns reward, terminated, info."""
+        self.time_step += 1
+
+        if isinstance(actions, np.ndarray):
```

Comment:
remove torch requirement, all the elements in env should be `np.array` and python native scalar